### PR TITLE
[RHINENG-11199] Update script to parse correctly None values

### PR DESCRIPTION
**Pull request by r0x0d**
When we parse the values coming from environment variable, the `None` value is parsed as a string. This patch introduces a simple solution to deal with that and not append a wrong value to the command line switch and to the environment context.

[RHINENG-11199](https://issues.redhat.com/browse/RHINENG-11199)

### CODE REVIEW

File: tests/test_environment_variables.py [hunk 1]
```diff
@@ -43,28 +43,43 @@ def test_parse_environment_variables_empty(monkeypatch):
...
-            {"CONVERT2RHEL_ALLOW_UNAVAILABLE_KMODS": 0, "SCRIPT_MODE": "ANALYSIS"},
+            {"CONVERT2RHEL_ALLOW_UNAVAILABLE_KMODS": "0", "SCRIPT_MODE": "ANALYSIS"},
             {"SCRIPT_MODE": "ANALYSIS"},
         ),
         (
             {
-                "CONVERT2RHEL_OUTDATED_PACKAGE_CHECK_SKIP": 1,
-                "CONVERT2RHEL_ALLOW_UNAVAILABLE_KMODS": 0,
+                "CONVERT2RHEL_OUTDATED_PACKAGE_CHECK_SKIP": "1",
+                "CONVERT2RHEL_ALLOW_UNAVAILABLE_KMODS": "0",
```
Comment (andywaltlova) [inline tests/test_environment_variables.py:64]: Is there also test for the "None" value? I assume now the Paramus should take everything except falsy values and "None"

**Review by andywaltlova — COMMENTED** [review]
Just one comment, otherwise looks fine :)

File: tests/test_environment_variables.py (hunk 1 above)
Comment (r0x0d) [reply tests/test_environment_variables.py:64]: Currently, not... We expect that it should be 0 or 1. Not sure if that's needed to add a case for None values for those. 

I guess that advisor-backend is enforcing the value to be either 0 or 1, so anything that is not in that rage should be discarded.

File: convert2rhel_insights_tasks/main.py [hunk 2]
```diff
@@ -674,7 +674,13 @@ def prepare_environment_variables(env):
     :type env: dict[str, Any]
     """
     for variable, value in env.items():
-        if variable.startswith("CONVERT2RHEL_") and value == 0:
+        # We can pop out of context both the OPTIONAL_REPOSITORIES and
+        # ELS_DISABLED envs as they are not necessary for convert2rhel
+        # execution.
+        if variable in ("OPTIONAL_REPOSITORIES", "ELS_DISABLED"):
+            env.pop(variable)
```
Comment (bocekm) [inline convert2rhel_insights_tasks/main.py:681]: This condition can stay but it seems it doesn't need to be here. Leaving the vars there will cause no harm.

File: convert2rhel_insights_tasks/main.py [hunk 3]
```diff
@@ -701,8 +707,15 @@ def run_convert2rhel(env):
     if not bool(els_disabled):
```
Comment (bocekm) [inline convert2rhel_insights_tasks/main.py:707]:
```suggestion
    if not els_disabled:
```
The json.loads already returns a boolean.

**Review by bocekm — APPROVED** [review]

File: convert2rhel_insights_tasks/main.py (hunk 3 above)
Comment (r0x0d) [reply convert2rhel_insights_tasks/main.py:707]: Ah, that's true. I can remove this in the next patch. It won't harm the script

File: convert2rhel_insights_tasks/main.py (hunk 2 above)
Comment (r0x0d) [reply convert2rhel_insights_tasks/main.py:681]: I don't want those extra properties being propagated to the env context for convert2rhel. As clean as we can make it, I want it to be.

Comment (bocekm) [reply convert2rhel_insights_tasks/main.py:681]:
```suggestion
        # We can pop out any env var that is not necessary for convert2rhel execution
        if not variable.startswith("CONVERT2RHEL_"):
            env.pop(variable)
```
Ack. So what about making it more generic?
